### PR TITLE
Change GET plan to observe role (WIP)

### PR DESCRIPTION
**Pull request by micheal-w-wells**

### CODE REVIEW

File: src/router/controllers_v1/PlanController.js [hunk 1]
```diff
@@ -32,20 +33,49 @@ export default class PlanController {
     );
 
     try {
+
       const [plan] = await Plan.findWithStatusExtension(db, { 'plan.id': planId }, ['id', 'desc']);
       if (!plan) {
-        throw errorWithCode('Plan doesn\'t exist', 404);
+        throw errorWithCode('Plan doesn\'t exist o', 404);
```
Comment (micheal-w-wells) [inline src/router/controllers_v1/PlanController.js]: @calebissharp could use your eyeballs buddy.

I somehow get:
```
{
    "error": "There was a problem fetching the record. Error: Plan doesn't exist",
    "success": false
}
```
despite that:

```
grep -inHr "Plan does" src/*                                                     
src/libs/db2/model/plan.js:160:      throw errorWithCode('Plan doesn\'t exist', 404);                          
src/router/controllers_v1/PlanController.js:39:        throw errorWithCode('Plan doesn\'t exist o', 404); 
```

Note the extra `o` after `exist` to differentiate the two (for now). 
The match in https://github.com/bcgov/range-api/blob/5f515b782eb4cedd3e3109238d5daaca1e3a04b5/src/libs/db2/model/plan.js#L160 is for `createSnapshot`.

What am I missing here?

Comment (micheal-w-wells) [reply src/router/controllers_v1/PlanController.js]: All agreements in my zone also and yes i did a build.

Comment (calebissharp) [reply src/router/controllers_v1/PlanController.js]: @micheal-w-wells Are you running this locally? Does your local database have any plans?

Comment (micheal-w-wells) [reply src/router/controllers_v1/PlanController.js]: @calebissharp its a prod backup and I stole all the agreements for my zone

Comment (micheal-w-wells) [reply src/router/controllers_v1/PlanController.js]: but ya local

Comment (calebissharp) [reply src/router/controllers_v1/PlanController.js]: @micheal-w-wells Prod may not be the best database to work off of, unless you ran all those migrations against it.

Comment (micheal-w-wells) [reply src/router/controllers_v1/PlanController.js]: The mystery for me is why can I only get the error string that is in create snapshot.  I did run the migrations

**Comment by micheal-w-wells** [comment]
Addressed in other pr

**Pull request closed without merging**